src/internal/promutil/promutil.go

```diff
@@ -0,0 +1,79 @@
+// Package promutil contains utilities for collecting Prometheus metrics.
+package promutil
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	inFlightMetric = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "http_client_in_flight_requests",
+		Help: "A gauge of in-flight requests being made against an HTTP API, by client.",
+	}, []string{"client"})
+
+	requestCountMetric = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "http_client_requests_total",
+		Help: "A summary of requests made against an HTTP API, by client, status code, and request method.",
+	}, []string{"client", "code", "method"})
+
+	requestTimeMetric = promauto.NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
We might want to weight these buckets down to smaller increments, or eliminate them entirely (At least, I sure hope we're not seeing 60s+ requests very often)

Reply:
I think we have k8s watches that last for ~30 minutes.